hardware-security-module/scripts/generate-hsm-user.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+set -eo pipefail
+# script directory
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+# define the CA setup
+CA_HOST=localhost
+CA_URL=${CA_HOST}:7054
+
+TLS_CERT="${SCRIPT_DIR}/../../test-network/organizations/fabric-ca/org1/tls-cert.pem"
+
+LocateHsmLib() {
+  if [[ -n "${PKCS11_LIB}" && -f "${PKCS11_LIB}" ]]; then
+    echo "${PKCS11_LIB}"
+    return
+  fi
+
+  local POSSIBLE_LIB_LOC=( \
+    '/usr/lib/softhsm/libsofthsm2.so' \
+    '/usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so' \
+    '/usr/local/lib/softhsm/libsofthsm2.so' \
+    '/usr/lib/libacsp-pkcs11.so' \
+  )
+  for TEST_LIB in "${POSSIBLE_LIB_LOC[@]}"; do
+    if [ -f "${TEST_LIB}" ]; then
+      echo "${TEST_LIB}"
+      return
+    fi
+  done
+}
+
+HSM2_LIB=$(LocateHsmLib)
+[ -z "$HSM2_LIB" ] && echo No SoftHSM PKCS11 Library found, ensure you have installed softhsm2 && exit 1
+
+# create a softhsm2.conf file if one doesn't exist
+HSM2_CONF=$HOME/softhsm2.conf
+[ ! -f "$HSM2_CONF" ] && echo directories.tokendir = /tmp > "$HSM2_CONF"
+
+# Update the client config file to point to the softhsm pkcs11 library
+# which must be in $HOME/softhsm directory
+
+CLIENT_CONFIG_TEMPLATE=${SCRIPT_DIR}/../ca-client-config/fabric-ca-client-config-template.yaml
+CLIENT_CONFIG=${SCRIPT_DIR}/../ca-client-config/fabric-ca-client-config.yaml
+cp $CLIENT_CONFIG_TEMPLATE $CLIENT_CONFIG
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  sed -i '' s+REPLACE_ME_HSMLIB+"${HSM2_LIB}"+g $CLIENT_CONFIG
+else
+  sed -i s+REPLACE_ME_HSMLIB+"${HSM2_LIB}"+g $CLIENT_CONFIG
+fi
+
+# create the users, remove any existing users
+CRYPTO_PATH=$SCRIPT_DIR/../crypto-material/hsm
+[ -d "$CRYPTO_PATH" ] && rm -fr "$CRYPTO_PATH"
+
+# user passed in as parameter
+CAADMIN="admin"
+CAADMIN_PW="adminpw"
+HSMUSER=$1
+
+SOFTHSM2_CONF=$HSM2_CONF fabric-ca-client enroll -c $CLIENT_CONFIG -u https://$CAADMIN:$CAADMIN_PW@$CA_URL --mspdir "$CRYPTO_PATH"/$CAADMIN --csr.hosts example.com --tls.certfiles "${TLS_CERT}"
+! SOFTHSM2_CONF=$HSM2_CONF fabric-ca-client register -c $CLIENT_CONFIG --mspdir "$CRYPTO_PATH"/$CAADMIN --id.name "$HSMUSER" --id.secret "$HSMUSER" --id.type client --caname ca-org1 --id.maxenrollments 0 -m example.com -u https://$CA_URL --tls.certfiles "${TLS_CERT}" && echo user probably already registered, continuing
```

Comment:
Apparently I have been bitten by the golang "error handling narrative" idioms.

This line would greatly be improved by realigning the arguments to the enrollment, one-per-line with `\` line endings for readability.

More importantly, the `!` and `&& echo user probably did something previously` is understandable but a bit reckless.  A preferred approach here would be to make the call (after formatting to avoid bleeding eyes), and check the $? error code with a little tweak here and there to the `set -eo pipefail` to not automatically exit the script if the user had previously registered the `HSMUser`.

CA client distinguishes between the error types in the return code.  It would be nice to just see this in the script rather than try to infer what `!` means in a script with the && ... at the end.   (or do something || true) 

E.g.: 
```
+ fabric-ca-client register -c /Users/jkneubuh/github.com/sapthasurendran/fabric-samples/hardware-security-module/scripts/../ca-client-config/fabric-ca-client-config.yaml --mspdir /Users/jkneubuh/github.com/sapthasurendran/fabric-samples/hardware-security-module/scripts/../crypto-material/hsm/rcaadmin --id.name HSMUser --id.secret HSMUser --id.type client --caname org1-ca --id.maxenrollments 0 -m vcap.me -u https://org1-ca.vcap.me:443 --tls.certfiles /Users/jkneubuh/github.com/sapthasurendran/fabric-samples/hardware-security-module/scripts/../../test-network-k8s/build/cas/org1-ca/tlsca-cert.pem
2022/07/13 12:47:45 [INFO] Configuration file location: /Users/jkneubuh/github.com/sapthasurendran/fabric-samples/hardware-security-module/ca-client-config/fabric-ca-client-config.yaml
2022/07/13 12:47:45 [INFO] TLS Enabled
2022/07/13 12:47:45 [INFO] TLS Enabled
Error: Response from server: Error Code: 74 - Identity 'HSMUser' is already registered

+ echo user probably already registered, continuing
```